src/main/java/com/amazon/opendistroforelasticsearch/sql/parser/SqlParser.java

```diff
@@ -95,6 +95,10 @@ public Select parseSelect(MySqlSelectQueryBlock query) throws SqlParseException
 
         select.setWhere(whereParser.findWhere());
 
+        if (select.getWhere() != null) {
```

Comment:
I'm looking if any other code path needs this fix. Should we patch `parseDelete` too?